[GEP-21] Handle Seed IPFamilies (IPv6)

breuerfelix opened the pull request:
**How to categorize this PR?**
/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This is part of GEP-21. Implementation of the local IPv6 Seeda

**Which issue(s) this PR fixes**:
Part of #7051 

**Special notes for your reviewer**:
Loki is not yet IPv6 ready until https://github.com/grafana/dskit/pull/185 is merged.

**Release note**:
```feature developer
Gardener's local setup now supports bootstrapping a Seed with IPv6 single-stack networking using `make gardener-up IPFAMILY=ipv6`. See the [documentation](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md) for more detailed steps.
```

Commit:
Replace decentralized FeatureGates with one centralized FeatureGate

Signed-off-by: Niclas Schad <niclas.schad@stackit.de>

Commit:
Regenerate certificate for gardener-admission-controller webhook

Commit:
Enable featureGate IPv6SingleStack and load featuresGates before validating SeedConfig

Commit:
Only filter envoyfilter via port number

Currently filtering by name restricts traffic to IPv4 only. By filtering
via port number this restriction is resolved and all traffic is respected.

Commit:
Filter config files via ipFamily

breuerfelix committed:
Deploy IPv6 ready gardenlet

breuerfelix committed:
Set IPv6 CIDRs via skaffold

breuerfelix committed:
Set IPv6 IPs for local development setup

breuerfelix committed:
Update Documentation for IPv6 local development

Signed-off-by: Felix Breuer <fbreuer@pm.me>

rfranzke commented on charts/images.yaml:
```diff
@@ -173,7 +173,7 @@ images:
 - name: grafana
   sourceRepository: github.com/grafana/grafana
-  repository: grafana/grafana
+  repository: eu.gcr.io/gardener-project/3rd/grafana/grafana
```
Can you extend https://github.com/gardener/gardener/blob/676ec251d1381932940be9b31f8e9e781aa62534/charts/images.yaml#L1-L5 with information that Docker Hub should be avoided because it doesn't support IPv6?

rfranzke commented on pkg/features/features.go:
```diff
@@ -111,6 +112,9 @@ const (
 )
 
+// DefaultFeatureGate TODO: Comment
+var DefaultFeatureGate = utilfeature.DefaultMutableFeatureGate
```
I would suggest to drop this commit, please. So far, we preferred individual feature gates per component/binary, and there should be no good reason to change this now. At least not as part of this PR. If you still think it's valuable, please trigger a separate/dedicated discussion about it via opening an according issue.

breuerfelix replied:
This PR does not work if this commit is not included here unfortunately...
The problem: while validating the config.yaml for the gardenlet, we use the Seed validation since there is a seed object in the config that configures the gardenlet. In this config there is also the definition on which features gates should be activated and which not.
The validation of the Seed object requires the SingleStack feature gate to be activated but we haven't yet activated the gate since we are still validating the config.yaml, thus we haven't applied it.
Would you suggest creating another PR on which this one here depends?

rfranzke replied:
Hold your horses for now - as said above, I don't even know whether we want this. I believe it requires a discussion first.
Where is this validation code that you mentioned which requires all this? I tried to find it in the PR/commits, but wasn't successful. Maybe we can find a solution without it or at least a compromise to not block this PR.

breuerfelix replied:
Okay i will open a discussion for this :)
The respective code is here https://github.com/gardener/gardener/blob/931b37c85f6ea6ec5c5e898a3e9c4235b628db7d/pkg/apis/core/validation/utils.go#L192 and it got introduced in an earlier PR where all IPv6 types got introduced (https://github.com/gardener/gardener/pull/7288/files).
The validation code was meant to be used in the apiserver to validate seed objects, but as stated, the gardenlet got a seed object inside its startup configuration.

rfranzke replied:
Would it be an option to 
- move this function to https://github.com/gardener/gardener/tree/master/pkg/utils/validation/cidr (possibly the `cidr` package could be renamed to `networking` or similar)
- pass a `bool` to the function describing whether IPv6 single stack should be allowed?

Then the respective components could pass their feature gates accordingly to the `bool` argument?

timebertt replied:
There were some misunderstandings here: with the change we use a central feature gate map shared across all component, but the components still register their individual feature gates.
This doesn't change that feature gates are only available on the components for which they are relevant.
However, we can now share code that checks a certain feature gate's state across components, which wasn't possible before.

I explained this in a code comment in a new commit :)

rfranzke commented on hack/local-development/generate-webhook-ca-bundle:
```diff
@@ -0,0 +1,55 @@
+prompt        = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = req_distinguished_name
```
The formatting seems broken here

rfranzke commented on hack/local-development/generate-webhook-ca-bundle:
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
```
Add `.sh` suffix to filename

rfranzke commented on example/20-componentconfig-gardenlet.yaml:
```diff
@@ -125,6 +125,7 @@ featureGates:
   ForceRestore: false
   DefaultSeccompProfile: true
   CoreDNSQueryRewriting: true
+  IPv6SingleStack: true
```
I don't think this should be enabled in this example file.

```suggestion
  IPv6SingleStack: false
```

breuerfelix replied:
It is enabled in the apiserver example per default aswell, thats why i made it here accordingly. Should i disable it in both cases?

rfranzke replied:
For the API server, it is only used for validation purposes and only when the `ipFamily=IPv6` was set. Hence, it didn't cause "harm" so far. Perhaps it's indeed reasonable to disable it everywhere?

rfranzke commented on cmd/gardenlet/app/options.go:
```diff
@@ -72,6 +73,11 @@ func (o *options) complete() error {
+	// Add feature flags
+	if err := features.DefaultFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
+		return err
+	}
```
Why was this moved to `validate()`? This sounds wrong - nothing is validated here. I would suggest to move it back. Now it's also inconsistent compared to how the feature gates are handled in the other binaries.

breuerfelix replied:
I described the problem in the other comment. TLDR; during the validation of the seed object inside the config, we check if the feature gate is enabled or not.

rfranzke commented on example/gardener-local/gardenlet/values.yaml:
```diff
@@ -36,6 +36,7 @@ config:
     CopyEtcdBackupsDuringControlPlaneMigration: true
     DefaultSeccompProfile: true
     CoreDNSQueryRewriting: true
+    IPv6SingleStack: true
```
Same here - doesn't this mean IPv6 single stack will be enabled in all local Gardener environments?

breuerfelix replied:
yes, but this just means you can provide IPv6 addresses and Gardener validates them. IPv4 does also work.

rfranzke replied:
https://github.com/gardener/gardener/pull/7561#discussion_r1119256147 is related. Maybe yes, maybe the feature gate will lead to behavioural changes in the future. Perhaps it's safer to disable it by default even in the local environments.

rfranzke commented on pkg/operation/botanist/component/istio/charts/istio/istio-ingress/templates/envoy-filter.yaml:
```diff
@@ -36,7 +36,6 @@ spec:
     match:
       context: GATEWAY
       listener:
-        name: 0.0.0.0_{{ .targetPort }}
```
cc/FYI @ScheererJ 
similar changes:
- https://github.com/gardener/gardener/pull/7561/commits/ad692ab6ec369c43d8e5216a5372610331479d70#diff-54653513c9d30c5e6bd47a82351cdb3b2e67a04f6e29275ca70e3db381c9f582
- https://github.com/gardener/gardener/pull/7561/commits/ad692ab6ec369c43d8e5216a5372610331479d70#diff-c54bf915e534f9bdb05b19b4c9be5e5a3aa8540703eb8426b052d72128032214

ScheererJ replied:
`portNumber` should be sufficient indeed. Otherwise we would need to duplicate the section with both an IPv4 and an IPv6 address. This approach seems better.

rfranzke commented on example/20-componentconfig-gardener-admission-controller-ipv6.yaml:
```diff
@@ -0,0 +1,35 @@
+logFormat: json
+server:
+  webhooks:
+    bindAddress: "::"
```
Why not `"[::]"` here as well?

breuerfelix replied:
somehow the webhook gets handled differently and this will throw the error `listen tcp: address [[::]]:2721: missing port in address` because it adds the braces on its own, not like the other servers

rfranzke replied:
Why is this? 😕 That's a bit odd and probably something we should clean up.

breuerfelix replied:
We just concatenated host and port together. I updated the PR and use the `net` package which the controller-runtime also uses to put host and port together for the webhook server. Now they all use `::` as the bind address :)

rfranzke commented on example/gardener-local/gardenlet/values-ipv6.yaml:
```diff
@@ -0,0 +1,36 @@
+        users:
+        - name: gardenlet-bootstrap
+          user:
+            token: 07401b.f395accd246ae52d
```
Is this needed? I don't see anything IPv6-specific in here?

breuerfelix replied:
`server: https://kubernetes.default.svc:443` this line changes. Not the internal kubernetes Service gets used for access. cc @einfachnuralex

ScheererJ replied:
Why was it necessary to change from `https://gardener-local-control-plane:6443` to `https://kubernetes.default.svc:443`?

breuerfelix replied:
We debugged this issue a little bit more yesterday and it basically boils down to this kind issue i opened https://github.com/kubernetes-sigs/kind/issues/3114 
We don't really know why it does not resolve the kind host when ipv6 is enabled as it works in ipv4 perfectly.

rfranzke replied:
Thanks, please add a comment with a link to this issue so that we don't forget about that.

rfranzke commented on example/provider-local/seed-kind/local-ipv6/seed.yaml:
```diff
@@ -0,0 +1,47 @@
+apiVersion: core.gardener.cloud/v1beta1
```
Would it be possible to work with patches to prevent copying the entire manifest? Like https://github.com/gardener/gardener/tree/master/example/provider-local/seed-kind2/local?

rfranzke commented on hack/local-development/common/helpers:
```diff
@@ -142,8 +142,15 @@ get_host_address () {
   local ip_route=""
   local ip_address=""
 
+  # TODO: SC2235
```
Sorry?

breuerfelix replied:
cc @nschad

nschad replied:
I forgot to remove it, the line below the TODO can be optimized by grouping instead of many subshells.
Docs: https://www.shellcheck.net/wiki/SC2235

Not really necessary you can remove the comment if you want.

rfranzke replied:
😵‍💫 Either fix it now or remove it, I doubt that anybody will pick this TODO up otherwise 😉

rfranzke commented:
/ok-to-test

breuerfelix committed:
Use net package to join host with port

Signed-off-by: Felix Breuer <fbreuer@pm.me>

rfranzke commented on charts/images.yaml:
```diff
@@ -259,7 +260,7 @@ images:
 - name: kubernetes-dashboard
   sourceRepository: github.com/kubernetes/dashboard
-  repository: kubernetesui/dashboard
+  repository: mirror.gcr.io/kubernetesui/dashboard
```
As discussed in the other PR, we probably should not rely on the GCR mirror and rather push these images to the `gardener` registry.

timebertt replied:
tracked in https://github.com/gardener/ci-infra/issues/619

timebertt replied:
The image switch can be done in a dedicated PR to unblock this one.
Continuing with docker hub images is fine – at least for now. The local kind seed basically has dual-stack internet access and is able to pull images from docker hub.
Later on, when working on the IPv6 shoot workers this won't work, but we will finish https://github.com/gardener/ci-infra/issues/619 by then probably.

Hence, I will drop the related changes from this PR.

timebertt commented:
/assign
I will take over the PR from @breuerfelix

timebertt committed:
Make `IPv6SingleStack` explanation clearer

timebertt committed:
Clean up gardenlet `bootstrapKubeconfig` values

Use the same `bootstrapKubeconfig` for all clusters with the in-cluster
kubernetes service except for the kind2 cluster.
With this, kind2 won't work with IPv6 single-stack, but we can live with
this for now.

timebertt committed:
Fix SC2235

timebertt committed:
Document purpose of central feature gate map

timebertt commented:
I resolved the remaining comments and all merge conflicts.
Though, I still need to iterate over this once more, clean some things up, and do a final test.
Until that is done:
/hold

timebertt commented:
/test pull-gardener-e2e-kind
/test pull-gardener-e2e-kind-ha-multi-zone-upgrade

timebertt committed:
Switch back to kind node's hostname

We tried to use `kubernetes.default.svc` instead of the kind node's hostname
as the garden cluster address for the gardenlet values.
This works from within the kind cluster itself also if it is IPv6 single-stack.
However, it doesn't work from within `ManagedSeeds`.
Hence, the managed seed e2e fails because gardenlet cannot register itself in the garden cluster.

timebertt committed:
Document requirement for IPv6 `localhost` entry in `/etc/hosts`

timebertt committed:
Introduce `garden.local.gardener.cloud` hostname that works everywhere

This replaces the use of kind container names to reach the garden cluster
and to reach the registry mirrors.
This works in
- the first and the second kind cluster
- in IPv4 and IPv6 kind clusters

timebertt committed:
Adapt `0.0.0.0/0` `NetworkPolicies`

timebertt committed:
Use `net.JoinHostPort` in all components

timebertt committed:
Drop `bindAddress=0.0.0.0` everywhere, drop unnecessary IPv6 switches

`net.Listen` listens on all available IP addresses if the hostname
(configured by `*bindAddress` fields) is omitted.
I.e., this works for both IP families without explicit configuration.
This simplifies a lot of config files and development scripts.

timebertt committed:
Make `bindAddress` chart values optional

timebertt commented:
/hold cancel
@rfranzke @ScheererJ thanks for your patience and all your feedback so far.
I've rebased the PR, addressed the feedback, introduced a workaround for https://github.com/kubernetes-sigs/kind/issues/3114, and simplified some things again (see the last 11 commits).

This is ready for another round of review now :)

timebertt committed:
Dump logs from containerd-configuration-local-setup service

timebertt committed:
Fix IP and cert handling in provider-local process-based setup

timebertt commented:
I did some final tests and had to fix some more stuff in provider-local.
With the latest changes, bringing up an IPv6 single-stack seed works with both the skaffold-based and process-based setup. :)

timebertt commented:
Rebase done :)

timebertt commented:
Rebased again :)

timebertt committed:
Add missing license header

timebertt commented:
@rfranzke @ScheererJ Could you kindly take another look before the rebase fun starts again? 😉

ScheererJ commented on example/20-componentconfig-gardenlet.yaml:
```diff
@@ -114,6 +114,7 @@ featureGates:
   CopyEtcdBackupsDuringControlPlaneMigration: true
   DefaultSeccompProfile: true
   CoreDNSQueryRewriting: true
+  IPv6SingleStack: true
```
I am a bit confused why @rfranzke's comment was resolved while the feature gate is still enabled here. It looks like the result of the discussion was the opposite. Can you please clarify?

timebertt replied:
That's my fault. I missed adding the results of my out-of-band discussion with @rfranzke on github.

The reasoning is that the `IPv6SingleStack` feature gate doesn't change apiserver's/gardenlet's default behavior. It only allows the creation of IPv6 seeds and shoots. The behavior only changes when creating such objects.
We agreed on enabling it by default in the local setup so that devs don't have to modify the configuration to get started with IPv6. If devs don't want to use IPv6, nothing changes.
Hence, I clarified the implications of enabling the feature gate in `feature_gate.md`.

ScheererJ commented on example/gardener-local/gardenlet/values.yaml:
```diff
@@ -36,6 +36,7 @@ config:
     CopyEtcdBackupsDuringControlPlaneMigration: true
     DefaultSeccompProfile: true
     CoreDNSQueryRewriting: true
+    IPv6SingleStack: true
```
I am a bit confused why @rfranzke's comment was resolved while the feature gate is still enabled here. It looks like the result of the discussion was the opposite. Can you please clarify?

ScheererJ commented on example/gardener-local/controlplane/auth-webhook-kubeconfig-local-ipv6.yaml:
```diff
@@ -0,0 +1,16 @@
+- name: garden
+  cluster:
+    insecure-skip-tls-verify: true
+    server: https://[fd00:10:2::1000]/webhooks/auth/seed
```
Is this supposed to be the real address? Why not use the IPv6 documentation range instead?

timebertt replied:
Yes, this is the IP of the `gardener-admission-controller` service in the namespace.
I added the same comment as in the other file to make this easier to understand.

ScheererJ commented on example/gardener-local/controlplane/auth-webhook-kubeconfig-skaffold-ipv6.yaml:
```diff
@@ -0,0 +1,17 @@
+  cluster:
+    insecure-skip-tls-verify: true
+    # This is the IP of the `gardener-admission-controller` service in the `garden` namespace.
+    server: https://[fd00:10:2::1000]/webhooks/auth/seed
```
Is this supposed to be the real address? Why not use the IPv6 documentation range instead?

ScheererJ commented on example/gardener-local/gardenlet/values-ipv6.yaml:
```diff
@@ -0,0 +1,38 @@
+          ipFamilies:
+          - IPv6
+          pods: fd00:10:3::/56
+          services: fd00:10:4::/112
```
Any particular reason for using fairly readable/easy ranges for everything except for the `nodes`? Would it not make sense to have the `nodes` range readable/easy as well?

timebertt replied:
The `nodes` CIDR is determined by the range of the `kind` docker network. This is configured by kind and we don't want to make it more complicated by requiring devs to recreate the `kind` network with a "readable" range.

ScheererJ replied:
Would it not be possible to automate the creation of the `kind` network and still achieve nice IP addresses?
I completely agree that additional manual steps should be omitted wherever possible, but the `kind` cluster is created by "us". Therefore, we should be able to configure it properly.
WDYT?

timebertt replied:
Fair enough. We also tell developers to recreate the kind network if it gets created with the "wrong" IPv4 CIDR (ref https://github.com/gardener/gardener/pull/6935). This could be automated as well.
I will give it a try and add a new commit to this PR to tackle both problems.

timebertt replied:
Done in a23065d9514f69f2b42f3bc0ef859fd05ffa05d8, let me now what you think :)

ScheererJ commented on example/provider-local/seed-kind/local-ipv6/patch-seed.yaml:
```diff
@@ -0,0 +1,11 @@
+    services: fd00:10:2::/112
+    shootDefaults:
+      pods: fd00:10:3::/56
+      services: fd00:10:4::/112
```
Again, it is unclear why we end up with hand-crafted, readable ranges for everything except to `nodes`. Could you please clarify?

ScheererJ commented on hack/kind-up.sh:
```diff
@@ -114,6 +114,48 @@ if [[ "$KUBECONFIG" != "$PATH_KUBECONFIG" ]]; then
+  cut -d/ -f2 |\
+  xargs -I {} docker exec {} sh -c "echo $garden_cluster_ip garden.local.gardener.cloud >> /etc/hosts"
+
+# Inject garden.local.gardener.cloud into coredns config (after ready plugin, before kubernetes plugin)
+kubectl -n kube-system get configmap coredns -ojson | \
+  yq '.data.Corefile' | \
+  sed '0,/ready.*$/s//&'" \n\
+    hosts { \n\
+      $garden_cluster_ip garden.local.gardener.cloud \n\
+      fallthrough \n\
+    } \
+"'/' | \
+  kubectl -n kube-system create configmap coredns --from-file Corefile=/dev/stdin --dry-run=client -oyaml | \
+  kubectl -n kube-system patch configmap coredns --patch-file /dev/stdin
+
+kubectl -n kube-system rollout restart deployment coredns
```
Would it not be better to simply adapt https://github.com/gardener/gardener/blob/master/charts/gardener/provider-local/extension/templates/coredns/custom-configmap.yaml? Just adding another override in the custom dns configuration feels like the straight forward way to do it. It would also remove the need to trigger the rollout.

timebertt replied:
Unfortunately, the custom coredns is not configured as the default DNS server and is hence only used by some of the pods, see the `dnsconfig` webhook of provider-local: https://github.com/gardener/gardener/blob/fd601cb65732d7c5efa35e4736e064a1f21ce4f5/pkg/provider-local/webhook/dnsconfig/add.go#L72-L78

The motivation for the new `garden.local.gardener.cloud` hostname was to make it resolvable/reachable from everywhere. Using the custom coredns doesn't fulfill that requirement, unfortunately.

ScheererJ replied:
Everywhere meaning everywhere within the cluster. Right?
If I resolve `garden.local.gardener.cloud` on my machine I will not get a proper response.

Nevertheless, you are correct in that the custom coredns config does not solve the issue. We just have too many dns servers in the local setup ;-) ...

timebertt replied:
Correct. Currently, it does not resolve on the host machine, because it was not needed so far.

ScheererJ commented on hack/ci-common.sh:
```diff
@@ -39,7 +39,7 @@ export_artifacts() {
 
       # systemd units
-      for unit in cloud-config-downloader kubelet containerd; do
+      for unit in cloud-config-downloader kubelet containerd containerd-configuration-local-setup; do
```
Could you please share how this is related to this PR? I do not see the connection.

timebertt replied:
It's not strictly related to this PR.
However, I debugged an e2e test failure and was wondering whether the new `containerd-configuration-local-setup` was properly executed on the shoot worker node. Then I noticed, that we don't dump the logs yet, so I added it :)

ScheererJ reviewed (changes requested):
Thanks for the changes so far. I have a few questions/comments.

timebertt committed:
Add missing comment for `gardener-admission-controller` `clusterIP` usage

timebertt commented:
@ScheererJ I replied to your questions, I hope things are clearer now. Can you take another look please? :)

timebertt commented:
.... aaaaaaaand rebased 😁

rfranzke commented on cmd/gardenlet/app/app.go:
```diff
@@ -131,11 +133,7 @@ func NewCommand() *cobra.Command {
 }
 
 func run(ctx context.Context, cancel context.CancelFunc, log logr.Logger, cfg *config.GardenletConfiguration) error {
-	// Add feature flags
-	if err := gardenletfeatures.FeatureGate.SetFromMap(cfg.FeatureGates); err != nil {
-		return err
-	}
```
Nit: Doesn't this `SetFromMap` invocation still better belong here than in `validate`?

timebertt replied:
Calling `SetFromMap` in `run` is too late. The config file validation will complain about the IPv6 `seedConfig` if the `IPv6SingleStack` feature gate is not enabled (via `SetFromMap`) beforehand.
I improved the comment to make this clearer and performed the change consistently across all component entry points.

rfranzke reviewed (commented):
Looks good

timebertt committed:
Consistently set feature gates in `validate`

timebertt committed:
Manage kind network ourselves

timebertt commented:
@rfranzke @ScheererJ I rebased and addressed your suggestions :)

timebertt committed:
Replace `jq` in `kind-up.sh` with `yq`

timebertt commented:
This one keeps being a pain to rebase. @rfranzke @ScheererJ I would highly appreciate a relieving approval 😄

rfranzke commented on cmd/gardener-controller-manager/app/options.go:
```diff
@@ -68,6 +69,11 @@ func (o *options) complete() error {
+	// Add feature flags before validating the config file, as they might influence the validation logic.
+	if err := features.DefaultFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
+		return err
+	}
```
Shouldn't we rather move this to the `app.go` before the `opts.complete()` is called? It seems weird/wrong to have it here in `validate()`, don't you agree?

timebertt replied:
> Calling `SetFromMap` in `run` is too late. The config file validation will complain about the IPv6 `seedConfig` if the `IPv6SingleStack` feature gate is not enabled (via `SetFromMap`) beforehand.

ref https://github.com/gardener/gardener/pull/7561#discussion_r1155738146

rfranzke replied:
I am not suggesting to call it in `run()` but before `opts.complete()` which is called here: https://github.com/gardener/gardener/blob/7b30c1147001d7ceaab6b822ad171be26e4cd12f/cmd/gardener-controller-manager/app/app.go#L62

timebertt replied:
Ah sorry, I misread your comment.
You're right., `SetFromMap` should rather be done in `complete()` than in `validate()`.
Calling it before `complete()` is too early, the config is only decoded in `complete()`.

I will move the `SetFromMap` calls to `complete()`.

timebertt replied:
Done in a new commit, tell me if you like it :)

timebertt committed:
Move `DefaultFeatureGate.SetFromMap` to `options.complete`

ScheererJ reviewed (commented):
/lgtm

rfranzke commented:
/approve

nschad commented:
🥳 🕺 💃 

Good job everyone!

jguipi commented:
It seems like since that PR was merged, there are a few errors that are generated by the remote-local-setup : 

while trying to start the Gardelet ( `make start-gardenlet SEED_NAME=local` ) 
<img width="1540" alt="image" src="https://user-images.githubusercontent.com/11049349/231553218-8da6c43c-046d-477a-a9fd-38568ab17981.png">

Putting the emphasis on : 
```
panic: failed during bootstrapping: incorrect node network specified in seed configuration (cluster node="172.18.0.2" vs. config="fd00:10::/64")
```

And while trying to run ( ` KUBECONFIG=${KUBECONFIG%:*} USER=root make start-extension-provider-local` ). I'm getting the following.

<img width="1313" alt="image" src="https://user-images.githubusercontent.com/11049349/231554464-75833710-817d-4ec6-82fd-d273fc99becb.png">

Can you look at that potential regression ?
